product/create-product.ts

```diff
@@ -0,0 +1,45 @@
+import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb';
+import { DynamoDB } from '@aws-sdk/client-dynamodb';
+
+const TABLE_NAME = process.env.TABLE_NAME || '';
+const STOCK_TABLE_NAME = process.env.STOCK_TABLE_NAME || '';
+
+const PRIMARY_KEY = process.env.PRIMARY_KEY || '';
+const STOCK_PRIMARY_KEY = process.env.STOCK_PRIMARY_KEY || '';
+
+const db = DynamoDBDocument.from(new DynamoDB());
+
+export const createProduct = async (event: any = {}) => {
+  if (!event.body) {
+    return { statusCode: 400, body: 'invalid request, you are missing the parameter body' };
+  }
+  const item = typeof event.body == 'object' ? event.body : JSON.parse(event.body);
+
+  const id = String(Date.now() + ~~(Math.random() * 1000))
+
+  const { stock, ...restFields } = item;
+
+  const params = {
+    TableName: TABLE_NAME,
+    Item: {
+        [PRIMARY_KEY]: id,
+        ...restFields
```

Comment:
> return error 400 status code if product data is invalid

1. if `restFields` contain `id` then the `[PRIMARY_KEY]: id` will be overwritten. It results in a bug when existing products may be updated, instead of creating a new product.
2. not only the expected fields, but any field from the request is written to the database.
3. fields types are not validated.
4. no validation to check that `count` is integer and non-negative. no validation to check that `price` is integer and positive.